pkg/dns/serviceresolver.go

```diff
@@ -239,7 +244,7 @@ func (b *ServiceResolver) Records(dnsName string, exact bool) ([]msg.Service, er
 		glog.V(4).Infof("Answered %s:%t with %#v", dnsName, exact, services)
 		return services, nil
 	}
-	return nil, nil
+	return nil, errNoSuchName
```

Comment:
I think that in order to get conformant behavior with upstream, we have to do this.  They materialize to etcd, so the answers to all your questions is the same (if they can't get the data, they write nothing to etcd).  So the failure modes being different than Kube would in theory be not conformant.